src/data-workspace/data-value-mutations/data-value-mutations.js

```diff
@@ -98,11 +98,24 @@ export function useUploadFileDataValueMutation({ deId, cocId }) {
 /** The `mutate` function returned by this hook expects no arguments */
 export function useDeleteDataValueMutation({ deId, cocId }) {
     const dataValueParams = useDataValueParams({ deId, cocId })
-    const mutationFn = useDeleteDataValueMutationFunction(dataValueParams)
+    const mutationFn = useDeleteDataValueMutationFunction()
 
     return useSharedDataValueMutation({
+        mutationKey: mutationKeys.delete(dataValueParams),
         dataValueParams,
         mutationFn,
         optimisticUpdateFn: optimisticallyDeleteDataValue,
     })
 }
+
+export function useSetDataValueMutationDefaults(queryClient) {
+    queryClient.setMutationDefaults(mutationKeys.file(), {
+        mutationFn: useUploadFileDataValueMutationFunction(),
+    })
+    queryClient.setMutationDefaults(mutationKeys.update(), {
+        mutationFn: useSetDataValueMutationFunction(),
+    })
+    queryClient.setMutationDefaults(mutationKeys.delete(), {
+        mutationFn: useDeleteDataValueMutationFunction(),
+    })
+}
```

Comment:
is this following the rules of hooks? 😅

Reply:
Yup, it might look weird since normally you would have it above. But there's no conditional heres, they're just nested in the argument. I could move them above though. 

```
const uploadFileMuateFn =  useUploadFileDataValueMutationFunction()
etc...
```

Reply:
> is this following the rules of hooks? sweat_smile

I'm wondering the same. It's a bit odd to me that we set defaults every time a consumer of `useSetDataValueMutationDefaults` re-renders. I suppose it works technically right now, but conceptually it seems wrong to me and more frail than doing this in a `useEffect`, which is meant to be used for side-effects like these.

Reply:
Yeah, should probably add this in an effect. Should probably do the same with the setup in `useQueryClient`. Right now this is only called once, and that component is never re-render - so it works fine, but I agree - we shouldn't really rely on that.

Reply:
That's true. I'll leave it up to you how you want to proceed. In case you want to keep it as is, please add a TODO comment and a jira issue so we track this potential issue somewhere